File: server/src/main/java/ee/rsx/kata/bank/loans/validation/SsnValidationResultDTO.java
package ee.rsx.kata.bank.loans.validation;

import static ee.rsx.kata.bank.loans.validation.ValidationStatus.INVALID;

public record SsnValidationResultDTO(String ssn, ValidationStatus status) {

  public static SsnValidationResultDTO okResultWith(String ssn) {
    return new SsnValidationResultDTO(ssn, ValidationStatus.OK);
  }

  public static SsnValidationResultDTO invalidResultWith(String ssn) {
    return new SsnValidationResultDTO(ssn, INVALID);
  }
}

File: server/loans/api/src/main/kotlin/ee/rsx/kata/bank/loans/validation/ssn/SsnValidationResultDTO.kt
package ee.rsx.kata.bank.loans.validation.ssn

@JvmRecord
data class SsnValidationResultDTO(val ssn: String, val status: ValidationStatus) {
  companion object {
    fun okResultWith(ssn: String): SsnValidationResultDTO {
      return SsnValidationResultDTO(ssn, ValidationStatus.OK)
    }

    fun invalidResultWith(ssn: String): SsnValidationResultDTO {
      return SsnValidationResultDTO(ssn, ValidationStatus.INVALID)
    }
  }
}
